src/requests.rs

```diff
@@ -0,0 +1,257 @@
+use std::collections::HashMap;
+
+// User request
+#[derive(Eq,Debug)]
+pub struct Request {
+    method : String,
+    route : String,
+    headers : HashMap<String, String>,
+}
+
+// Server response
+#[derive(Eq,Debug)]
+pub struct Response {
+    status : i32,
+    content_type : String,
+    body : String,
+    headers : Option<HashMap<String, String>>,
+}
+
+// Allows for equality comparisons between requests/responses
+impl PartialEq for Request {
+    fn eq(&self, other: &Request) -> bool {
+        self.method == other.method
+        && self.route == other.route
+        && self.headers == other.headers
+    }
+}
+
+impl PartialEq for Response {
+    fn eq(&self, other: &Response) -> bool {
+        self.status == other.status
+        && self.content_type == other.content_type
+        && self.body == other.body
+        && self.headers == other.headers
+    }
+}
+
+// Methods available for Request and Response objects. Uses builder pattern.
+impl Request {
+
+    // Creates a new request object with default values
+    pub fn new() -> Request {
+	    let empty_header = HashMap::new();
+        let request = Request { method: "".to_string(),
+	                            route: "".to_string(),
+		                        headers: empty_header
+	    };
+	    request
+    }
+
+    /*
+    Takes the raw request string and transforms it into a Request object.
+    */
+    pub fn get_request_obj(mut self, request : &str) -> Request {
```

Comment:
can we rename this to 'from(mut self, request : &str) -> Request' to be keeping with Rust patterns?